### PR TITLE
fix: [CI-9556]: Add acl log-service

### PR DESCRIPTION
**Pull request by smjt-h**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56575)

### CODE REVIEW

**Comment by smjt-h** [comment]
trigger uts

**Comment by smjt-h** [comment]
trigger gobuild

**Comment by smjt-h** [comment]
trigger uts

**Comment by smjt-h** [comment]
trigger uts

File: product/log-service/handler/middleware.go [hunk 1]
```diff
@@ -126,57 +126,54 @@ func AuthMiddleware(config config.Config, ngClient, aclClient *client.HTTPClient
...
-				}
-			} else {
-				inputToken := r.Header.Get(authHeader)
-				if inputToken == "" {
-					inputToken = r.FormValue(authHeader)
+						WithField("accountID", accountID).
+						Errorln("middleware: failed to validate access control")
+					//Donot return here, as we need to check for old approach (x-api-key or X-harness-Token). this is for release backward compatibility. Will need to remove this later.
+				} else {
+					validated = true
```
Comment (vistaarjuneja) [inline product/log-service/handler/middleware.go:143]: can you confirm that validated should be true here?

File: product/log-service/handler/middleware.go [hunk 2]
```diff
@@ -126,57 +126,54 @@ func AuthMiddleware(config config.Config, ngClient, aclClient *client.HTTPClient
...
-					WriteBadRequest(w, errors.New("no token in header"))
-					return
-				}
-				// accountID in token should be same as accountID in URL
-				secret := []byte(config.Auth.LogSecret)
-				login := authcookie.Login(inputToken, secret)
-				if login == "" || login != accountID {
-					writeError(w, errors.New(fmt.Sprintf("operation not permitted for accountID: %s", accountID)), 403)
-					return
+			//Fallback to old approach (x-api-key or X-harness-Token) if not validatd by access control
```
Comment (vistaarjuneja) [inline product/log-service/handler/middleware.go:147]: Nit: fix spelling -> validated. And add a space after //

File: product/log-service/handler/middleware.go (hunk 1 above)
Comment (smjt-h) [reply product/log-service/handler/middleware.go:143]: yes, 
If there is no error then it should be validated true

**Comment by smjt-h** [comment]
trigger uts

File: product/log-service/handler/middleware.go (hunk 2 above)
Comment (smjt-h) [reply product/log-service/handler/middleware.go]: done

**Review by vistaarjuneja — APPROVED** [review]

**Pull request merged**